### PR TITLE
pkp/pkp-lib#1707 add supplementary file support for the native import and export

### PR DESCRIPTION
**Pull request by bozana**
S. https://github.com/pkp/pkp-lib/issues/1707

### CODE REVIEW

File: plugins/importexport/native/filter/NativeXmlArticleFilter.inc.php [hunk 1]
```diff
@@ -81,6 +81,9 @@ function handleChildElement($n, $submission) {
 			case 'artwork_file':
 				$this->parseSubmissionFile($n, $submission);
 				break;
+			case 'supplementary_file':
+				$this->parseSubmissionFile($n, $submission);
+				break;
```
Comment (NateWr) [inline plugins/importexport/native/filter/NativeXmlArticleFilter.inc.php]: The two `case` statements for `artwork_file` and `supplementary_file` can be combined into one.

File: plugins/importexport/native/filter/NativeXmlArtworkFileFilter.inc.php [hunk 2]
```diff
@@ -83,7 +84,7 @@ function handleChildElement($node, $stageId, &$submissionFiles) {
 				$submissionFiles[count($submissionFiles)-1]->setPermissionTerms($node->textContent);
 				break;
 			default:
-				return parent::handleChildElement($node, $stageId, $submissionFiles);
+				return parent::handleChildElement($node, $stageId, $fileId, $submissionFiles);
```
Comment (NateWr) [inline plugins/importexport/native/filter/NativeXmlArtworkFileFilter.inc.php]: `return` here can be removed, I think...

**Comment by bozana** [comment]
hmmm... why is this failing now... ?

**Comment by bozana** [comment]
@asmecher, what is happening with tests now? They are becoming more and more crazy... :-(

**Comment by bozana** [comment]
Ha, I think they get scared of me :-DDD
@NateWr, could you take a look at/review those additional commits? I couldn't reproduce your error, but if you would send me your DB...

File: controllers/grid/issues/form/IssueForm.inc.php [hunk 3]
```diff
@@ -65,7 +65,8 @@ function fetch($request) {
 		}
 
 		// Cover image preview
-		$coverImage = $this->issue->getCoverImage();
+		$coverImage = null;
+		if ($this->issue) $coverImage = $this->issue->getCoverImage();
```
Comment (NateWr) [inline controllers/grid/issues/form/IssueForm.inc.php]: A neat little one-liner for this is:

```
$coverImage = $this->issue ? $this->issue->getCoverImage() : null;
```

Neither way's better. Just sharing. :)

**Pull request merged**